pyswervedrive/swervemodule.py

```diff
@@ -228,24 +230,27 @@ def set_velocity(self, vx, vy):
     def steer_enc_offset(self):
         return int(self.steer_enc_offset_entry.getDouble(0))
 
-    @property
-    def current_azimuth(self):
-        """Return the current azimuth from the controller setpoint in radians."""
-        setpoint = self.steer_motor.getClosedLoopTarget()
-        return float(setpoint - self.steer_enc_offset) / self.STEER_COUNTS_PER_RADIAN
-
     @property
     def current_measured_azimuth(self):
         """Return the azimuth of the wheel as measured by the encoder."""
-        pos = self.steer_motor.getSelectedSensorPosition(0)
-        return constrain_angle(float(pos - self.steer_enc_offset)
-                               / self.STEER_COUNTS_PER_RADIAN)
+        return self.measured_azimuth
 
     @property
     def current_speed(self):
         """Return the current speed of the module's wheel"""
-        wheel_vel = self.drive_motor.getSelectedSensorVelocity(0)
-        return wheel_vel / self.drive_velocity_to_native_units
+        return self.wheel_vel
+
+    def update_odometry(self):
+        # self.wheel_pos = 0
+        # self.measured_azimuth = 0
+        # self.wheel_vel = 0
+        # return
+        drive_pos, drive_vel = self.drive_motor.getQuadratureSensor()
+        self.wheel_vel = drive_vel / self.drive_velocity_to_native_units
```

Comment:
Given that `wheel_vel` is only used for the odometry stuff from what I can tell, I don't think we need to read the drive velocity any more. In this case, we should probably stop using the APIs that "directly" read the encoders, since they seem to not behave properly with `TalonSRX#setSensorPhase()`.

Reply:
We probably shouldn't read the quadrature sensor anyway; the Talons don't send that as frequently as the closed-loop sensor readings.